### PR TITLE
Noetic devel

### PR DESCRIPTION
**Pull request by davefeilseifer**
these changes will allow pr2-simulator to compile in 20.04

### CODE REVIEW

File: pr2_gazebo_plugins/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 cmake_minimum_required(VERSION 2.8.12)
 project(pr2_gazebo_plugins)
 
+set(CXXFLAGS -std=c++17)
```
Comment (v4hn) [inline pr2_gazebo_plugins/CMakeLists.txt]:
```suggestion
set(CMAKE_CXX_STANDARD 17)
```

File: pr2_gazebo_plugins/CMakeLists.txt [hunk 2]
```diff
@@ -13,12 +14,14 @@ else()
...
+
+
 # Depend on system install of Gazebo and SDFormat
 find_package(gazebo REQUIRED)
-pkg_check_modules(SDF sdformat REQUIRED)
+pkg_check_modules(SDF sdformat9 REQUIRED)
 #find_package(PCL REQUIRED)
 find_package(Boost REQUIRED COMPONENTS thread)
-find_package(orocos_kdl)
+find_package(orocos_kdl HINTS "/usr/share/pkgconfig")
```
Comment (v4hn) [inline pr2_gazebo_plugins/CMakeLists.txt]: What? Why is there a hint for pkgconfig in a cmake `find_package` call?

File: pr2_gazebo_plugins/package.xml [hunk 3]
```diff
@@ -71,7 +71,5 @@
   <run_depend>message_generation</run_depend>
 
   <!-- temporary hack for orocos kdl until it's packaged better upstream -->
-  <build_depend>orocos_kdl</build_depend>
-  <run_depend>orocos_kdl</run_depend>
```
Comment (v4hn) [inline pr2_gazebo_plugins/package.xml:75]:
```suggestion
  <build_depend>liborocos-kdl</build_depend>
  <run_depend>liborocos-kdl</run_depend>
```

**Review by v4hn — CHANGES_REQUESTED** [review]
As some of these fixes are not compatible with melodic/kinetic anymore, I'd suggest to merge the fixes into a new branch `noetic-devel` and release from there.

**Comment by davefeilseifer** [comment]
I'm not sure I'm able to create a new branch with the access I have, is it possible for one of you to do so?

**Comment by v4hn** [comment]
> I'm not sure I'm able to create a new branch with the access I have, is it possible for one of you to do so?

Assuming you can merge pull-requests I would expect you can also push new branches.
I just pushed a noetic-devel branch as a copy of the current kinetic-devel branch and retargeted the merge request

File: pr2_gazebo_plugins/CMakeLists.txt (hunk 2 above)
Comment (davefeilseifer) [reply pr2_gazebo_plugins/CMakeLists.txt]: agreed, committed change

**Review by davefeilseifer — COMMENTED** [review]
changes made in outside commits

**Comment by robogeekcanada** [comment]
Thank you @davefeilseifer. I was able to run in ROS noetic with the changes proposed. If anyone is interested, I captured the steps how to do installation in this tutorial: https://github.com/robogeekcanada/pr2_noetic

This may help someone until this branch is fully committed.

**Pull request closed without merging**